src/Tribe/Models/Post_Types/Event.php

```diff
@@ -96,56 +97,73 @@ protected function build_properties( $filter ) {
 			}
 
 			if ( Dates::is_valid_date( $filter ) ) {
-				$week_start = Dates::build_date_object( $filter, $timezone );
-				// Sunday is 0.
-				$week_start_day = (int) get_option( 'start_of_week' );
-				$offset         = (int) $week_start->format( 'N' ) >= $week_start_day
-					? $week_start_day
-					: $week_start->format( 'N' ) - $week_start_day;
-
-				$week_start->setISODate( (int) $week_start->format( 'o' ), (int) $week_start->format( 'W' ), $offset );
-				$week_end = clone $week_start;
-				// 7 days later the week ends.
-				$week_end->add( new DateInterval( 'P7D' ) );
-				// Inclusive in respect to the start, exclusive to the end.
-				$starts_this_week  = $week_start <= $start_date_object && $start_date_object < $week_end;
-				$ends_this_week    = $week_start <= $end_date_object && $end_date_object < $week_end;
-				$happens_this_week = $week_start <= $end_date_object && $start_date_object <= $week_end;
+				list( $week_start, $week_end ) = Dates::get_week_start_end( $filter );
+
+				$the_start = $start_date_object;
+				$the_end   = $end_date_object;
+
+				// Take into account the timezone settings.
+				if ( Timezones::is_mode( Timezones::SITE_TIMEZONE ) ) {
+					// Move the event to the site timezone.
+					$the_start = $the_start->setTimezone( $site_timezone );
+					$the_end   = $the_end->setTimezone( $site_timezone );
+				}
+
+				$week_start_ymd = (int) $week_start->format( 'Ymd' );
+				$week_end_ymd   = (int) $week_end->format( 'Ymd' );
+				$the_start_ymd  = (int) $the_start->format( 'Ymd' );
+				$the_end_ymd    = (int) $the_end->format( 'Ymd' );
+
+				$starts_this_week  = $week_start_ymd <= $the_start_ymd && $the_start_ymd <= $week_end_ymd;
+				$ends_this_week    = $week_start_ymd <= $the_end_ymd && $the_end_ymd <= $week_end_ymd;
+				$happens_this_week = $week_start_ymd <= $the_end_ymd && $the_start_ymd <= $week_end_ymd;
+
+				/*
+				 * A day "crosses the EOD cutoff time" if the end is after the EOD cutoff of the start.
+				 * Here we look just for a boolean.
+				 */
+				$cross_day = tribe_end_of_day( $the_start->format( 'Y-m-d' ) ) < $the_end->format( 'Y-m-d H:i:s' );
```

Comment:
Can we use the `Date_Utils` constants for these formats?

Reply:
For all intents and purposes, this is a "service" format.
A few lines above that I build the `_ymd` variables using the `Ymd` format since we do not have a `Dates` constant for that, we leave that alone. But the concept is the same.